addons/purchase_stock/tests/test_create_picking.py

```diff
@@ -313,3 +313,65 @@ def test_04_mto_multiple_po(self):
         purchase_order_2.picking_ids.button_validate()
 
         self.assertEqual(sum(customer_picking.move_lines.mapped('reserved_availability')), 100.0, 'The total quantity for the customer move should be available and reserved.')
+
+    def test_04_rounding(self):
+        """ We set the Unit(s) rounding to 1.0 and ensure buying 1.2 units in a PO is rounded to 1.0
+            at reception.
+        """
+        uom_unit = self.env.ref('uom.product_uom_unit')
+        uom_unit.rounding = 1.0
+
+        # buy a dozen
+        po = self.env['purchase.order'].create(self.po_vals)
+
+        po.order_line.product_qty = 1.2
+        po.button_confirm()
+
+        # the move should be 1.0 units
+        move1 = po.picking_ids.move_lines[0]
+        self.assertEqual(move1.product_uom_qty, 1.0)
+        self.assertEqual(move1.product_uom.id, uom_unit.id)
+        self.assertEqual(move1.product_qty, 1.0)
+
+        # edit the so line, buy 2.4 units, the move should now be 2.0 units
+        po.order_line.product_qty = 2.0
```

Comment:
Shouldn't the test assign `2.4` instead?

Reply:
Too bad I saw this comment too late.